### PR TITLE
Added support for building ZIP distributions

### PR DESCRIPTION
**Pull request by AyaanShah2204**
- Run `pyinstaller .\apps\stable_diffusion\studio_bundle.spec -y` to build a single-directory application for SHARK
- Run `python .\build_tools\bundled_app_to_zip.py` to build a zip file in the main directory

onefile exe applications are still created in the same way they were before: `pyinstaller .\apps\stable_diffusion\shark_sd.spec`

Reason:
`--onefile` and `--onedir` are commands that pyinstaller accepts when it is pointed to a python script. These commands dictate how a spec file is created. Once the spec file is created, it will produce either a folder or an exe file depending on how the spec file was configured
Hence, if we want our build commands to point to spec files (because it allows us to use hidden imports and pyinstaller hooks without writing an unwieldy shell command) we need separate spec files for exe and zip

### CODE REVIEW

File: build_tools/bundled_app_to_zip.py
Comment (powderluv) [inline build_tools/bundled_app_to_zip.py:1]: lets drop this file. We can run this from the command line.

File: apps/stable_diffusion/studio_bundle.spec
Comment (powderluv) [inline apps/stable_diffusion/studio_bundle.spec:1]: can we move all of the common parts of both .spec files to a shark_studio_imports.py and then lets import into both .spec files.

File: build_tools/bundled_app_to_zip.py
Comment (AyaanShah2204) [reply build_tools/bundled_app_to_zip.py:1]: Done! We can use `Compress-Archive` to zip the bundle from nightly build scripts.

File: apps/stable_diffusion/studio_bundle.spec
Comment (AyaanShah2204) [reply apps/stable_diffusion/studio_bundle.spec:1]: Moved all the heavy looking stuff :)

**Review by powderluv — APPROVED** [review]

**Pull request merged**